### PR TITLE
Make reads and writes to sockets silent with the @-operator

### PR DESCRIPTION
**Pull request by Kekos**
If the MQTT daemon closes the connection before `MQTTClient` sends a keepalive ping the socket ends up broken and PHP issues a notice. Some frameworks halts the execution on all errors which makes it impossible to catch the `DataTransferException`  and reconnect.

This commit makes all `fread` and `fwrite` completely silent and hands over to the already built in error handling in `MQTTClient`.

### CODE REVIEW

**Comment by Namoshek** [comment]
I'm not convinced this solution works for all applications. Laravel applications for example have a built-in error handler which does ignore the error suppressing operator `@` and will throw anyway. 

The standard way to check for closed sockets seems to be using `fread($socket, 1)` which will return `false` for closed sockets (not a PHP warning) and an empty string if the connection is still open (and if no data has been sent by the server). The problem with this approach is it requires the non-blocking mode of the socket to be used and currently this library allows using both modes (although I don't know why you'd use the blocking one). It also seems problematic to me that using `fread` could actually fetch some incoming data when testing the socket for its state, which in return would be missing in `readFromSocket()` later.

To sum it up, `fread($sock)` should never throw a PHP warning but `fwrite($sock)` does. So I'd suggest we only suppress the warning of `fwrite` to avoid suppressing critical issues unwantedly, at least until we find a better solution (or PHP provides a better one :smile:).

**Comment by Kekos** [comment]
Thank you for an interesting clarifying about `fread()`!
